src/software/network_log_listener_main.cpp

```diff
@@ -31,39 +31,89 @@ void logFromNetworking(TbotsProto::RobotLog log)
 
     LOG(level) << "[ROBOT " << log.robot_id() << " " << LogLevel_Name(log.log_level())
                << "]"
-               << "[" << log.file_name() << ":" << log.line_number()
-               << "]: " << log.log_msg() << std::endl;
+               << "[" << log.file_name() << ":" << log.line_number() << "] "
+               << log.created_timestamp().epoch_timestamp_seconds() << ": "
+               << log.log_msg();
 }
 
 int main(int argc, char **argv)
 {
     // load command line arguments
-    auto args           = std::make_shared<NetworkLogListenerMainCommandLineArgs>();
-    bool help_requested = args->loadFromCommandLineArguments(argc, argv);
+    struct CommandLineArgs
+    {
+        bool help                     = false;
+        std::string interface         = "";
+        int channel                   = 0;
+        std::vector<int> filtered_ids = {};
+    };
+
+    CommandLineArgs args;
+    boost::program_options::options_description desc{"Options"};
+
+    desc.add_options()("help,h", boost::program_options::bool_switch(&args.help),
+                       "Help screen");
+    desc.add_options()("interface",
+                       boost::program_options::value<std::string>(&args.interface),
+                       "Which network interface to listen for messages from");
+    desc.add_options()("channel", boost::program_options::value<int>(&args.channel),
+                       "Multicast channel to listen on connect to");
+    desc.add_options()(
+        "filtered_ids",
+        boost::program_options::value<std::vector<int>>(&args.filtered_ids)->multitoken(),
+        "Robot IDs to filter out of the logs");
+
+    boost::program_options::variables_map vm;
+    boost::program_options::store(parse_command_line(argc, argv, desc), vm);
+    boost::program_options::notify(vm);
+
+    if (args.help)
+    {
+        std::cout << desc << std::endl;
+        return 0;
+    }
+
+    if (!vm.count("interface"))
+    {
+        std::cerr << "A network interface must be specified to listen on!" << std::endl;
+        return 0;
+    }
 
     auto logWorker               = g3::LogWorker::createLogWorker();
     auto colour_cout_sink_handle = logWorker->addSink(
         std::make_unique<ColouredCoutSink>(false), &ColouredCoutSink::displayColouredLog);
     g3::initializeLogging(logWorker.get());
 
-    if (!help_requested)
-    {
-        int channel           = args->getChannel()->value();
-        std::string interface = args->getInterface()->value();
+    // Callback which filters out the logs from robots with filtered_id
+    auto robot_log_callback = [args](TbotsProto::RobotLog log) {
+        if (std::find(args.filtered_ids.begin(), args.filtered_ids.end(),
```

Comment:
shouldn't the check for "==" (i.e. return early if the robot log id isn't found in the filtered arg list?

aka the find returns args.filtered_id.end()

Reply:
So the `filtered_ids` arg is used to filter out robots that you do NOT want to see logs from (Say Im testing on robot 1 and you're testing on robot 2 and 3, I will have `--filtered_ids 2 3`). So the `find` searches over the `filtered_ids` list to see if the ID of the current log is in it. If it is (i.e. `!= args.filtered_ids.end()`), then it returns early.

Reply:
Okay that sounds good. I've flipped it to `selected_ids`. If nothing is passed, it will show all robot logs

Reply:
that's a great idea